src/cmps/GamingCard.tsx

```diff
@@ -1,39 +1,84 @@
-import colors from "../assets/constants/colors";
+import { useEffect, useRef, useState } from "react";
 import {
   StyledGamingCard,
-  ImageContainer,
   AvatarContainer,
-  StyledProgress,
+  ImageContainer,
   PowerLevelContainer,
   PokemonImage,
+  StyledProgress,
 } from "../styles/StyledGamingCard";
 import Typography from "../styles/Typography";
 import { Pokemon } from "../interfaces/Pokemon";
+import colors from "../assets/constants/colors";
 import { PowerLevel } from "../styles/StyledPokemonCard";
 import { SPACING } from "../assets/constants/spacings";
 import { TypographyType } from "../enums/TypographyEnum";
 
 interface GamingCardProps {
   pokemon: Pokemon;
+  hp: number;
+  isUser: boolean;
+  activeTurn: boolean;
+  hit: number;
+  isFightClicked: boolean;
 }
 
-const GamingCard = ({ pokemon }: GamingCardProps) => {
+const GamingCard = ({
+  pokemon,
+  hp,
+  isUser,
+  activeTurn,
+  hit,
+  isFightClicked,
+}: GamingCardProps) => {
+  const [animatedHP, setAnimatedHP] = useState(hp);
+  const [isAnimating, setIsAnimating] = useState(false);
+  const prevHpRef = useRef(hp);
+
+  useEffect(() => {
+    if (hit > 0) {
+      setIsAnimating(true);
+      const interval = setInterval(() => {
+        setAnimatedHP((prevAnimatedHP) => {
+          const newHP = Math.max(prevAnimatedHP - 1, hp);
+          if (newHP === hp) {
+            clearInterval(interval);
+            setIsAnimating(false);
+          }
+          return newHP;
+        });
+      }, 30);
```

Comment:
can you use useMemo instead of useEffect?

Reply:
i think useEffect is more suitable to this situation no? its a side effect and and the decreased HP is always a different calculation